alacritty/src/display/mod.rs

```diff
@@ -652,11 +648,19 @@ impl Display {
             self.window.set_resize_increments(PhysicalSize::new(cell_width, cell_height));
         }
 
-        // Resize PTY.
-        pty_resize_handle.on_resize(new_size.into());
+        // Resize when terminal when its dimensions have changed.
+        if self.size_info.screen_lines() != new_size.screen_lines
+            || self.size_info.columns() != new_size.columns()
+        {
+            // Resize PTY.
+            pty_resize_handle.on_resize(new_size.into());
 
-        // Resize terminal.
-        terminal.resize(new_size);
+            // Resize terminal.
+            terminal.resize(new_size);
+
+            // Resize damage tracking.
+            self.damage_tracker.resize(new_size.screen_lines(), new_size.columns());
+        }
 
         // Check if dimensions have changed.
         if new_size != self.size_info {
```

Comment:
We already check if the dimensions have changed further down here, is there a reason to check twice?

Reply:
They are not equivalent, because we could change width without changing dimensions, like you the change resulted only in the padding change.

Reply:
Fair enough though that seems extremely unlikely to happen in practice.